ibis/pandas/tests/test_udf.py

```diff
@@ -225,49 +250,107 @@ def my_corr2(lhs, **kwargs):
+    expr = t2.mutate(rolled=my_mean(t2.b).over(window))
+    result = expr.execute().sort_values(['key', 'a'])
+    expected = df2.sort_values(['key', 'a']).assign(
+        rolled=lambda df: df.groupby('key')
+        .b.rolling(3, min_periods=1)
+        .mean()
+        .reset_index(level=0, drop=True)
+    )
+    tm.assert_frame_equal(result, expected)
+
+
+def test_udaf_window_interval():
     df = pd.DataFrame(
-        {
-            'a': np.arange(4, dtype=float).tolist()
-            + np.random.rand(3).tolist(),
-            'b': np.arange(4, dtype=float).tolist()
-            + np.random.rand(3).tolist(),
-            'key': list('ddeefff'),
-        }
+        collections.OrderedDict(
+            [
+                (
+                    "time",
+                    pd.date_range(
+                        start='20190105', end='20190101', freq='-1D'
+                    ),
+                ),
+                ("key", [1, 2, 1, 2, 1]),
+                ("value", np.arange(5)),
+            ]
+        )
     )
+
     con = ibis.pandas.connect({'df': df})
     t = con.table('df')
-    expr = times_two(add_one(t.a))
-    result = expr.execute()
-    expected = df.a.add(1.0).mul(2.0)
-    tm.assert_series_equal(expected, result)
+    window = ibis.trailing_range_window(
+        ibis.interval(days=2), order_by='time', group_by='key'
+    )
+
+    expr = t.mutate(rolled=my_mean(t.value).over(window))
+
+    result = expr.execute().sort_values(['time', 'key']).reset_index(drop=True)
+    expected = (
+        df.sort_values(['time', 'key'])
+        .set_index('time')
+        .assign(
+            rolled=lambda df: df.groupby('key')
+            .value.rolling('2D', closed='both')
+            .mean()
+            .reset_index(level=0, drop=True)
+        )
+    ).reset_index(drop=False)
+
+    tm.assert_frame_equal(result, expected)
+
 
+def test_multiple_argument_udaf_window():
+    # PR 2035
 
-def test_udaf_window():
-    @udf.reduction(['double'], 'double')
-    def my_mean(series):
-        return series.mean()
+    @udf.reduction(['double', 'double'], 'double')
+    def my_wm(v, w):
+        return np.average(v, weights=w)
 
     df = pd.DataFrame(
         {
-            'a': np.arange(4, dtype=float).tolist()
+            'a': np.arange(4, 0, dtype=float, step=-1).tolist()
             + np.random.rand(3).tolist(),
             'b': np.arange(4, dtype=float).tolist()
             + np.random.rand(3).tolist(),
-            'key': list('ddeefff'),
+            'c': np.arange(4, dtype=float).tolist()
+            + np.random.rand(3).tolist(),
+            'd': np.repeat(1, 7),
+            'key': list('deefefd'),
         }
     )
     con = ibis.pandas.connect({'df': df})
     t = con.table('df')
     window = ibis.trailing_window(2, order_by='a', group_by='key')
-    expr = t.mutate(rolled=my_mean(t.b).over(window))
+    expr = t.mutate(
+        wm_b=my_wm(t.b, t.d).over(window), wm_c=my_wm(t.c, t.d).over(window)
```

Comment:
I assume that this will work if we use *different* windows? can you add a test for that, or if it doesn't work can you test the exception.

Reply:
Added